pkg/util/version/stream_test.go

```diff
@@ -40,52 +38,85 @@ func TestGetUpgradeStream(t *testing.T) {
 	stream44 := Stream{
 		Version: NewVersion(4, 4, 3),
 	}
+	stream45 := Stream{
+		Version: NewVersion(4, 5, 0),
+	}
 
 	for _, tt := range []struct {
-		name string
-		v    *Version
-		want Stream
-		err  error
+		name     string
+		v        *Version
+		want     *Stream
+		upgradeY bool
+		streams  []*Stream
 	}{
 		{
-			name: "upgrade when x.Y is lower than expected",
-			v:    NewVersion(4, 3, 17),
-			want: stream43,
+			name:     "upgrade when Y versions match and candidate Z (4.3.18) is greater",
+			v:        NewVersion(4, 3, 17),
+			upgradeY: false,
+			streams:  []*Stream{&stream43, &stream44},
+			want:     &stream43,
```

Comment:
style: in general, do `foo := &Foo{}` and define foo as a pointer type at the point of definition, then you don't spread `&`s all over the rest of the codebase